src/systems/implicit_system.C

```diff
@@ -675,21 +673,25 @@ void ImplicitSystem::assemble_residual_derivatives(const ParameterVector& parame
       // (R(p-dp) - R(p+dp)) / (2*dp)
 
       Number old_parameter = *parameters[p];
-      *parameters[p] -= deltap;
+
+      const Real delta_p =
+        TOLERANCE * std::max(std::abs(old_parameter), 1e-3);
```

Comment:
Just to make sure I'm clear, the `std::max` is there just to prevent `delta_p` from getting absurdly small? I guess if someone has an insanely small parameter, they need to change units?